docs/src/specs/device.md

```diff
@@ -48,17 +48,23 @@ each with an 8x8 physical grid.
 underlying physical hardware device.
 
 ```mlir
-#tt.device<#tt.grid<8x8, (d0, d1) -> (0, d0, d1)>, [0]>
+#tt.device<workerGrid = #tt.grid<8x8, (d0, d1) -> (0, d0, d1)>, meshShape = 1, chipIds = [0]>
 ```
 
 Let's break down what each of these attributes mean:
-- `#tt.grid<8x8, (d0, d1) -> (0, d0, d1)>`: This is a 2D logical grid with dim 8x8.
+- `workerGrid = #tt.grid<8x8, (d0, d1) -> (0, d0, d1)>`: This is a 2D logical grid with dim 8x8.
   It's followed by an affine map `(d0, d1) -> (0, d0, d1)` that provides a mapping
   from the logical grid to the physical grid.  In this case, the logical grid is the same
   as the physical grid, so the mapping is the identity function. The logical
   grid can have any rank, but the physical mapping is always 3D, with the first
   being the chip index, followed by the 2D physical core index within the chip.
-- `[0]`: This is a list of chip indices.  These chip indices directly reference
+- `meshShape = 1`: A shape provided as part of the `DeviceAttr` constructor that
+  describes the virtual layout of the chips with respect to each other. Note that
+  in a multi-chip system, this grid encapsulates the entire system's grid shape,
+  e.g. 8x16 grid could be made up of a 1x2 mesh of chips side-by-side. The mesh
+  attribute configures how the above grid/map attributes are created such that they
+  implement this mesh topology.
+- `chipIds = [0]`: This is a list of chip indices.  These chip indices directly reference
```

Comment:
Heads up, I intend to remove `chip_ids` from the constructor. It seems to be inconsistent between T3000 and TG -- It currently relies on forwarding physical device ids for T3000 and I'd like the user-side to only care about the logical grid.

Reply:
So is the intention that the main constructor for `MeshDevice` always opens all devices and then we can use `SubView` to partition the mesh?

Reply:
Not necessarily. I think there are two ways of looking at the mesh:
1. Open all the devices into a MeshDevice and use views to partition the mesh (what you described)
2. If a user only wants to open a 2x2, we shouldn't force them to open up the full 8x4. In this case, it would look like:


```py
# constructor: MeshDevice(mesh_shape, start_coord_yx = Coord(0,0))

# user code
mesh_device_0 = MeshDevice(mesh_shape=(2,2))
mesh_device_1 = MeshDevice(mesh_shape=(2,2), start=(0,2))
```

Reply:
Ok awesome, makes sense.  Is there an implied ordering, i.e. do you interpret the chip IDs in row major? Or do you do some more sophisticated topology analysis to figure out how to form a legal mesh?

Reply:
implied row-major. I'll sync with you offline.